tests/parsers/test_sqla.py

```diff
@@ -33,6 +35,15 @@ def test_parse_filter():
     assert isinstance(f, BinaryExpression)
     assert d == ww
 
+def test_parse_filter_with_alias():
+    ''' test a sqlalchemy filter parse with an aliased class '''
+    d = 'modela2.x > 5'
```

Comment:
Since `modela2` is an alias of `modela`, what happens if my filter contains both, e.g. `modela2.x > 5 or modela.x < 3`?  Or what happens when I specify only a unique field on `modela`, but both the original and aliased classes are supplied, i.e. `x > 5`?  We want to make sure these cases either resolves correctly or throw an informed error.  Can you write some tests for these cases?

Reply:
`modela2.x > 5 or modela.x < 3` should work as expected. The AliasedClass from SQL alchemy wraps the attributes so the field returned by `_get_field()` is different for ModelA.x and ModelA2.x. SQL alchemy will use the alias name in the sql it generates if one was provided (otherwise it generates it's own). I will add a test case for this.

If you ran a search for `x>5` with both ModelA and ModelA2 provided, it would use the first one that is listed first in the model classes list. This is the same (existing) behavior if ModelA and ModelB both have a field `X`. The filter method has a for loop through the models and breaks after the field is found with `type` and `ilike` attributes. I can add a test for this, but the real question is if you think the behavior should change? Rather than breaking from the for loop, could track the first field and raise an error if a second field is found.

Reply:
On the first point, that's good to know.  I hadn't realized aliased classes were somewhat independent of the original.  On the second point, ahh yeah that's right.  I think I'm ok with it grabbing from the first model it can find.  But I suppose the more informative thing to do would be to catch any multiples and raise an error informing that the name is too ambiguous, and to suggest they try `parent.name`.